server/storage/mvcc/index.go

```diff
@@ -194,14 +194,9 @@ func (ti *treeIndex) RangeSince(key, end []byte, rev int64) []revision {
 		return keyi.since(ti.lg, rev)
 	}
 
-	endi := &keyIndex{key: end}
 	var revs []revision
-	ti.tree.AscendGreaterOrEqual(keyi, func(item btree.Item) bool {
-		if len(endi.key) > 0 && !item.Less(endi) {
-			return false
-		}
-		curKeyi := item.(*keyIndex)
-		revs = append(revs, curKeyi.since(ti.lg, rev)...)
+	ti.visit(key, end, func(ki *keyIndex) bool {
```

Comment:
It will lead to deadlock, because it will acquire the read lock inside the visit method as well. We can add an `unsafeVisit` method to resolve it.